src/ts/core/features/filter-search.ts

```diff
@@ -0,0 +1,130 @@
+import {Feature, Settings} from '../settings'
+import {browser} from 'webextension-polyfill-ts'
+
+export const config: Feature = { // An object that describes new feature we introduce
+    id: 'filter_search',  // Feature id - any unique string would do
+    name: 'Filter Search',  // Feature name - would be displayed in the settings menu
+    //warning: 'Experimental feature',
+    enabledByDefault: true,
+}
+
+const checkSettingsAndSetupFilterSearchToggle = () => {
+    Settings.isActive('filter_search').then(active => {
+        toggleFilterSearch(active)
+    })
+}
+checkSettingsAndSetupFilterSearchToggle()
+
+browser.runtime.onMessage.addListener(async message => {
+    if (message === 'settings-updated') {
+        checkSettingsAndSetupFilterSearchToggle()
+    }
+})
+
+const toggleFilterSearch = (active: boolean) => {
+    const jsDoc = document;
+    if (active) {
+        jsDoc.addEventListener("click", filterSearch);
+    } else {
+        jsDoc.removeEventListener("click", filterSearch);
+    }
+}
+
+var debugMode = 0;
+
+function filterSearch(evt: MouseEvent)
+{
+    var evtTarget = (<HTMLElement>evt.target);
```

Comment:
Just curious for learning purposes.... why?

Reply:
Short version: it's easier to reason about behavior of `const` then about `let` and it's easier to reason about `let` then about `var`.  
See here on difference between `let` and `var`: https://stackoverflow.com/questions/762011/whats-the-difference-between-using-let-and-var  
`const` is like `let` but with the additional constraint that variables defined with it can't be reassigned